.travis.yml

```diff
@@ -19,26 +19,30 @@
 sudo: true
 dist: trusty
 language: python
-python:
-  - "3.6"
 env:
   global:
-    - DOCKER_COMPOSE_VERSION=1.20.0
-    - TRAVIS_CACHE=$HOME/.travis_cache/
+    - DOCKER_COMPOSE_VERSION=1.23.2
   matrix:
-    - TOX_ENV=py27-backend_mysql-env_docker
-    - TOX_ENV=py27-backend_sqlite-env_docker
-    - TOX_ENV=py27-backend_postgres-env_docker
-    - TOX_ENV=py35-backend_mysql-env_docker PYTHON_VERSION=3
-    - TOX_ENV=py35-backend_sqlite-env_docker PYTHON_VERSION=3
-    - TOX_ENV=py35-backend_postgres-env_docker PYTHON_VERSION=3
-    - TOX_ENV=py27-backend_postgres-env_kubernetes KUBERNETES_VERSION=v1.9.0
-    - TOX_ENV=py35-backend_postgres-env_kubernetes KUBERNETES_VERSION=v1.10.0 PYTHON_VERSION=3
-
+    - BACKEND=mysql ENV=docker
+    - BACKEND=postgres ENV=docker
+    - BACKEND=sqlite ENV=docker
+    - BACKEND=postgres ENV=kubernetes KUBERNETES_VERSION=v1.9.0
+    - BACKEND=postgres ENV=kubernetes KUBERNETES_VERSION=v1.10.0
+python:
+  - "3.6"
+  - "3.5"
```

Comment:
Do we still run the tests in docker under this? (I may answer my own question as I read the rest of the PR)

Reply:
My current approach (still testing it) is that yes - all the tests will be run in docker, same way they were run before. The main difference is that we prepare locally a final image that can be run for testing - from the current sources, utilising docker layers and registry as cache rather than .pip-cache adn .wheels directories mounted from the host. This is a bit cleaner and hopefully I can make it equally fast or even faster. 

I am still looking at some limitation and speed there so this might change if I find I cannot do it as good as with folder/sources/wheel/.pip_cache caching.

Reply:
Can we use experimental buildkit features? https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/experimental.md#run---mounttypecache Might work out as the best of both worlds?

Reply:
I actually tried to use buildkit already and I have a flag to enable it :) https://github.com/apache/airflow/pull/4543/files#diff-14bdd6aac779ac93a185193b26ac4fd3R76 - the problem with buildkit is that it's cache support does not work well with `docker build` command - buildkit uses different manifest and if you build+push using `docker build` such image is unusable for buildkit. You'd have to install and use buildkit directly (rather than rely on the experimental support in recent docker release)

Support for this is coming in the near future and we should revisit it then. Also in the future (soon) version of Docker there will be "cache" type mount that we migt use for that.

Reply:
I am resolving it for now. Buildkit is not yet usable for us (but will be soon and then we can switch - it's just a matter of flipping the flag in Dockerfile.